rotkehlchen/db/filtering.py

```diff
@@ -1062,3 +1082,144 @@ def make(
             ))
         filter_query.filters = filters
         return filter_query
+
+
+class NFTFilterQuery(DBFilterQuery):
+    @classmethod
+    def make(
+            cls,
+            order_by_rules: Optional[List[Tuple[str, bool]]] = None,
+            limit: Optional[int] = None,
+            offset: Optional[int] = None,
+            owner_addresses: Optional[List[str]] = None,
+            name: Optional[str] = None,
+            collection_name: Optional[str] = None,
+            ignored_assets_filter_params: Optional[Tuple[Literal['IN', 'NOT IN'], List[str]]] = None,  # noqa: E501
+    ) -> 'NFTFilterQuery':
+        filter_query = cls.create(
+            and_op=True,
+            limit=limit,
+            offset=offset,
+            order_by_rules=order_by_rules,
+        )
+        filter_query = cast('NFTFilterQuery', filter_query)
+        filters: List[DBFilter] = []
+        if owner_addresses is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='owner_address',
+                values=owner_addresses,
+            ))
+        if name is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='name',
+                search_string=name,
+            ))
+        if collection_name is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='collection_name',
+                search_string=collection_name,
+            ))
+        if ignored_assets_filter_params is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='identifier',
+                operator=ignored_assets_filter_params[0],
+                values=ignored_assets_filter_params[1],
+            ))
+        filter_query.filters = filters
+        return filter_query
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class MultiTableFilterQuery():
+    and_op: bool
+    filters: List[Tuple[DBFilter, str]]  # filter + table name for which to use it.
+
+    def prepare(self, target_table_name: str) -> Tuple[str, List[Any]]:
+        query_parts = []
+        bindings = []
+        filterstrings = []
+
+        for (fil, table_name) in self.filters:
+            if table_name != target_table_name:
+                continue
+
+            filters, single_bindings = fil.prepare()
+            if len(filters) == 0:
+                continue
+
+            operator = ' AND ' if fil.and_op else ' OR '
+            filterstrings.append(f'({operator.join(filters)})')
+            bindings.extend(single_bindings)
+
+        if len(filterstrings) != 0:
+            operator = ' AND ' if self.and_op else ' OR '
```

Comment:
Also actually since you use this multiple times, initialize it once at the start of the function

```suggestion
            operator = ' AND ' if self.and_op is True else ' OR '
```

Reply:
The first one is `fil.and_op` and this is `self.and_op`, so they are different. Or were you talking about something else?

Reply:
ah okay so then forget it. Neither can be abstracted.